File: Tourism-BE/src/main/java/com/gialai/tourism/models/entities/Notification.java
package com.gialai.tourism.models.entities;

import com.gialai.tourism.common.base.BaseEntity;
import com.gialai.tourism.enums.NotificationType;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "notifications")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Notification extends BaseEntity {

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private NotificationType type;

    @Column(nullable = false, length = 1000)
    private String message;

    @Column(nullable = false)
    private boolean isRead;

    @Column(name = "related_post_id")
    private String relatedPostId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "recipient_id", nullable = false)
    private Account recipient;
}

File: Tourism-FE/src/types/notification.ts
export type NotificationType =
  | 'POST_APPROVED'
  | 'POST_REJECTED'
  | 'POST_PENDING'
  | 'COMMENT'
  | 'SYSTEM';

export interface Notification {
  id: string;
  type: NotificationType;
  message: string;
  isRead: boolean;
  relatedPostId?: string;
  createdAt: string;
}
